### PR TITLE
IME without UI demo

### PR DESCRIPTION
**Pull request by lgvalle**
## Scope

This project consists in a custom IME (keyboard) without UI.

It consist in:
* `MainActivity` : A simple activity with an edit text to try out the IME
* `NovodaInputMethod` : A class extending InputMethodService that simply captures key events, show then using a `Toast` and calls `super`

## Goal
Demostrate the minimum requirements to create such IME and work as playground for future tests

## Screenshot

screenshot_1516642650

### CODE REVIEW

File: keyboarddemo/app/src/main/res/drawable-v24/ic_launcher_foreground.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (tasomaniac) [reply keyboarddemo/app/src/main/res/drawable-v24/ic_launcher_foreground.xml]: They are not mandatory right?

[the remark this answers is not in the document]
Comment (tasomaniac) [reply keyboarddemo/app/src/main/res/drawable-v24/ic_launcher_foreground.xml]: Also `ic_launcher` can be removed completely as it is not a must to have icons.

File: keyboarddemo/app/src/main/res/xml/method.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+
+<!-- The attributes in this XML file provide configuration information --><!-- for the Search Manager. -->
+
+<input-method xmlns:android="http://schemas.android.com/apk/res/android"
+  android:settingsActivity="com.blackcj.customkeyboard.ImePreferences"
```
Comment (tasomaniac) [inline keyboarddemo/app/src/main/res/xml/method.xml]: What is `com.blackcj.customkeyboard.ImePreferences` all about?

File: keyboarddemo/app/src/main/res/drawable-v24/ic_launcher_foreground.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rock3r) [reply keyboarddemo/app/src/main/res/drawable-v24/ic_launcher_foreground.xml]: No they are not mandatory :)

File: keyboarddemo/app/src/main/res/xml/method.xml (hunk 2 above)
Comment (rock3r) [reply keyboarddemo/app/src/main/res/xml/method.xml]: Copy-pasta leftover I assume? :)

Comment (lgvalle) [reply keyboarddemo/app/src/main/res/xml/method.xml]: Yeah :D

**Comment by lgvalle** [comment]
I pushed last minute just to have the code on Github, cleaning up now.

**Pull request merged**